lib/Fmt_ast.ml

```diff
@@ -605,11 +605,20 @@ and fmt_record_field c ?typ ?rhs ?(type_first = false) lid1 =
   in
   let fmt_type_rhs =
     match (typ, rhs) with
-    | Some t, Some r ->
-        if type_first then field_space $ fmt_type t $ fmt "@ " $ fmt_rhs r
-        else
+    | Some t, Some r -> (
+      match constraint_loc with
+      | Some (`Constraint_lid loc) ->
+          field_space
+          $ Cmts.fmt_before c loc ~pro:(Fmt.break 1 0) ~epi:noop
+          $ fmt_type t $ fmt "@ " $ fmt_rhs r $ Cmts.fmt_after c loc
+      | Some (`Constraint_exp loc) ->
+          field_space $ fmt "=@;<1 2>"
+          $ hvbox 0
+            @@ Cmts.fmt c loc
+                 (str "(" $ cbox 0 r $ str " " $ fmt_type ~parens:true t)
+      | None ->
```

Comment:
Why is this change needed ?

Reply:
Because https://github.com/ocaml/ocaml/pull/10555 also changed the location span used for Ppat_constraint/Pexp_constraint, so the formatting had to be updated accordingly.
Otherwise here is the diff I had:
```diff
 let _ =
   { (* a *) a (* a' *)= (* b *) b (* b' *)
-  ; (* c' *) (* c *) c: (* d *) d (* d' *) = (* e *) e (* e' *)
+  ; (* c' *)
+    (* c *) c: (* d *) d (* d' *) = (* e *) e
+    (* e' *)
   ; (* f *) f (* f' *)
   ; (* j *)
     (* g *) g (* g' *)= ((* h *) h (* h' *) : (* i *) i (* i' *))
     (* j' *) }
 
 let { (* a *) a (* a' *)= (* b *) b (* b' *)
-    ; (* c' *) (* c *) c: (* d *) d (* d' *) = (* e *) e (* e' *)
+    ; (* c' *)
+      (* c *) c: (* d *) d (* d' *) = (* e *) e
+      (* e' *)
     ; (* f *) f (* f' *)
     ; (* j *)
       (* g *) g (* g' *)= ((* h *) h (* h' *) : (* i *) i (* i' *))
```